parachain/primitives/core/src/lib.rs

```diff
@@ -143,13 +143,21 @@ pub enum AgentExecuteCommand {
 }
 
 impl AgentExecuteCommand {
+	fn index(&self) -> u8 {
+		match self {
+			AgentExecuteCommand::TransferToken { .. } => 0,
+		}
+	}
 	pub fn abi_encode(&self) -> Vec<u8> {
 		match self {
 			AgentExecuteCommand::TransferToken { token, recipient, amount } => {
 				ethabi::encode(&vec![
-					Token::Address(*token),
-					Token::Address(*token),
-					Token::Uint(U256::from(*amount)),
+					Token::Uint(self.index().into()),
+					Token::Bytes(ethabi::encode(&vec![Token::Tuple(vec![
```

Comment:
The Solidity side isn't expecting a tuple (struct) type though.

i.e `abi.decode(data, (address, address, uint128))` is equivalent to: 
```rust
ethabi::encode(&vec![Token::Address(*token), Token::Address(*recipient), Token::Uint(U256::from(*amount))]);
````

Reply:
I think implicitly it's encoded to the same thing because `(address, address, uint128)` is an implicit tuple. I have tested both versions and they work. The tuple and just a `vec![Tokens]`. I used the tuple version because it feels more correct. So I think this is more of style thing so which would you prefer?

Reply:
I prefer the original form.

We can't say for sure whether they encode to the same thing for all input parameter types, without more investigation and testing. Also `Token::Tuple` was introduced purely to encode structs, and we're not using structs in this case.